server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java

```diff
@@ -69,7 +69,9 @@ protected AbstractScopedSettings(
             final Settings settings,
             final Set<Setting<?>> settingsSet,
             final Set<SettingUpgrader<?>> settingUpgraders,
-            final Setting.Property scope) {
+            final Setting.Property scope,
+            final String prefix) {
+        this.logger = Loggers.getLogger(this.getClass(), prefix);
```

Comment:
I think it'd be better to allow subclasses to provide their own `logger` rather than just a `String`, because I don't think we should be changing the behaviour for the logging of cluster-level settings.